### ISSUE TITLE
 Configuring gitlab-ci.yml with staging server domain name

### ISSUE THREAD

**Issue by pli888** [gigascience/gigadb-website#283]
Hi @rija

The `with_new_cert_deploy` stage in [gitlab-ci.yml](https://github.com/gigascience/gigadb-website/blob/develop/.gitlab-ci.yml) has the domain name for the staging server hard-coded on line 244 when creating an SSL certificate using Let's Encrypt.

To replace the hard-coding, should we use the STAGING_HOME_URL variable in the GitLab CI/CD environment variables to provide the server domain name?

**Comment by rija** [comment]
Yes, that's correct. Sorry about that.

**Comment by pli888** [comment]
@rija 

Related to this issue is `gigadb-staging.pommetab.com` being present in the [gigadb.staging.http.conf](https://github.com/gigascience/gigadb-website/blob/develop/ops/configuration/nginx-conf/sites/gigadb.staging.http.conf) and [gigadb.staging.https.conf](https://github.com/gigascience/gigadb-website/blob/develop/ops/configuration/nginx-conf/sites/gigadb.staging.https.conf) files.

I was wondering if it is best to configure these nginx conf files to use the value of the STAGING_HOME_URL variable for setting the server_name directive?

**Comment by rija** [comment]
@pli888 

I did consider that when I created these files and then decided against putting variables in there.

My thinking was that these are config files and they are already keyed against a specific environment. Using environment specific name in them is therefore ok, and furthermore using variable interpolation when the file is used only in one place doesn't feel right.

Also these configuration files are not a place where every developer would need to change something to fit their local deployment, the ``gigadb.dev.http.conf`` is designed to work for everyone who use the Docker deploy while being reasonably faithful to production behaviour.

Similarly, after a period of trial and error, I expect the ``gigadb.staging.http.conf`` and ``gigadb.production.http.conf`` to stabilise and not change.

If it were desirable to not have staging and production hard-coded, there would need to be some research done as there are a few moving parts involved and some challenges, for example:
* At the moment, the Nginx config are baked directly in the Nginx Docker image using COPY
* Nginx does not have support for variables in its config
